server/db/driver/pg/internal/accounts.go

```diff
@@ -14,9 +14,13 @@ const (
 		pubkey BYTEA,
 		fee_address TEXT,
 		fee_coin BYTEA,
-		broken_rule INT2 DEFAULT 0
+		broken_rule INT2 DEFAULT 0,
+		forgiven_score INT4 DEFAULT 0
 		);`
 
+	AddAccountsForgivenScoreColumn = `ALTER TABLE %s
+		ADD COLUMN IF NOT EXISTS forgiven_score INT4 DEFAULT 0;`
```

Comment:
This fit in nicely before the N latest concept was implemented.  Unfortunately that it doesn't make sense any more as violations can fall off.   I hate to even consider the suggestion as well, but I'm afraid we are down to the need to forgive specific violations.  And while that works for matches, there's also preimage misses to consider.  Crap.

Reply:
Resolved this in https://github.com/decred/dcrdex/pull/700/commits/23e1ded14c80e85508de04d8f19eca5d38534b5f, eliminating forgiven_score and going with the `forgiven BOOL` column in matches.  I tried quite hard to avoid that, but it's just about the only way to do it short of a separate violations table. The status negation idea didn't work because NewlyMatched is 0.